L1Trigger/VertexFinder/src/VertexFinder.cc

```diff
@@ -1242,26 +1250,45 @@ namespace l1tVertexFinder {
     // 1 top PV from BRS Chosen:
     // vertices_.emplace_back(vertices.at(pv->second));
     // 1 top PV from Chris' network:
-    vertices_.emplace_back(vertices.at(nnChosenPV->first));
+    vertices_.emplace_back(vertices.at(nnChosenPV->first).z0());
 
     // #### Run track association: ####
     cout << "Track Association " << endl;
     tensorflow::Tensor inputAssoc(tensorflow::DT_FLOAT, {1, 4});  //Single batch of 4 values
 
+    // Bins used for track res feature to assoc network
+    std::vector<float> eta_bins = {0.0,0.2,0.4,0.6,0.8,1.0,1.2,1.4,1.6,1.8,2,2.2,2.4,3.0};
+    std::vector<float> res_bins  = {0.0,0.1,0.1,0.12,0.14,0.16,0.18,0.23,0.23,0.3,0.35,0.38,0.42,0.5,1};
+
     // loop over tracks
     uint trackIt(0);
     for (L1Track& track : fitTracks_) {
-      // Chris' Unquantised Network: deltaZ, normed_trk_pt, trk_MVA1, trk_over_eta_squared
-      float pT = track.pt() > 512.0 ? 1.0 : track.pt()/512.0;
-      // inputAssoc.tensor<float, 2>()(0, 0) = pT;
-      // inputAssoc.tensor<float, 2>()(0, 1) = float(track.MVA1()); //BDT track quality
-      // inputAssoc.tensor<float, 2>()(0, 2) = 5.76/(float(abs(track.eta()))*float(abs(track.eta())));
-      // inputAssoc.tensor<float, 2>()(0, 3) = float(abs(track.z0() - vertices_.at(0).z0()));
-
-      inputAssoc.tensor<float, 2>()(0, 0) = float(abs(track.z0() - vertices_.at(0).z0()));
-      inputAssoc.tensor<float, 2>()(0, 1) = pT;
-      inputAssoc.tensor<float, 2>()(0, 2) = float(track.MVA1()); //BDT track quality
-      inputAssoc.tensor<float, 2>()(0, 3) = 5.76/(float(abs(track.eta()))*float(abs(track.eta())));
+      auto& gttTrack = gttTracks_.at(trackIt);
+      if (track.z0() > 0.)
+          temp_z0 = track.z0() + 0.03;
+        else if (track.z0() < 0.)
+          temp_z0 = track.z0() - 0.03;
+
+      // Chris' Network: deltaZ, normed_trk_pt, trk_MVA1, trk_res
+      float pT = gttTrack.getTTTrackPtr()->getRinvBits();
+      float MVA = gttTrack.getTTTrackPtr()->getMVAQualityBits();
+      float Eta = gttTrack.getTTTrackPtr()->getTanlBits();
```

Comment:
We could copy the [code snippet from above](https://github.com/BenjaminRS/cmssw/pull/2/files#diff-8ffa95cfbf64f33de31fe0b2d2eec9ccdb3cedf579d194f844d04524bf3eb9cfR1126) to make it consistent in the same function.

Reply:
Fixed